### ISSUE TITLE
Read export QR code

### ISSUE THREAD

**Issue by olfek** [dim13/otpauth#32]
My backup process:

1. Take screenshot of Google Authenticator export QR code.
2. Use cable to move it to PC.
3. Need to load it into program ...

**Comment by dim13** [comment]
IIRC it is not possible to use screenshot function on GA's export view.

Use any QR reader program of your trust on your laptop through camera. Would strongly discourage of usage of any online service, at QR code(s) contains confidential information.

Feed decoded content to the app.

**Comment by olfek** [comment]
> IIRC it is not possible to use screenshot function on GA's export view.

It is, I did it earlier today.

> Use any QR reader program of your trust on your laptop through camera

There aren't any on Windows.

**Comment by dim13** [comment]
I'm not an expert on windows and no devices to test it there.

Quick search reveals however a buch of options available in official app store: https://apps.microsoft.com/store/search/qr%20reader

**Comment by olfek** [comment]
I meant there aren't any QR code scanning apps for Windows that I trust 😛

And the official Windows Camera app _can_ scan QR codes, but it doesn't like the GA export QR code.

![image](https://github.com/dim13/otpauth/assets/27283110/4ecca57d-83a0-4d63-b62e-f1ed5e5ed128)

**Comment by dim13** [comment]
This is way beyond my knowledge of windows. Your QR app tries to be smart and instead of simply displaying content tries to open the link.

Anyway, maybe this will help you: https://learn.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/platform-apis/aa767914(v=vs.85)?redirectedfrom=MSDN

Register new schema by adding
```HKEY_CLASSES_ROOT\otpauth-migtation\shell\open\command = "C:\Program Files\Alert\alert.exe" "alert:%1"```

Or something like that.

**Comment by olfek** [comment]
Yeah that might work, but earlier today I moved away from GA to Aegis (https://getaegis.app), which can export as plain text, so this issue isn't an issue for _me_ anymore, but maybe we can leave it open for others who have to:

1. Screenshot GA export QR code.
2. Scan QR code on mobile device.
3. Copy migration link.
4. Save migration link in a text file using favorite text editor app.
5. Clear mobile clipboard.
6. Move text file over to PC.
7. Copy migration link from text file.
8. Delete text file.
9. Run this program on copied migration link.
10. Clear PC clipboard.

**Comment by Toom7892** [comment]
![Screenshot_20240614-234112](https://github.com/dim13/otpauth/assets/172853479/5d23a6f9-7dbb-4342-bc76-ca59794e5efd)

**Comment by Toom7892** [comment]
HKEY_CLASSES_ROOT
   otpauth-migration
      (Default) = "URL:otpauth migration"
      URL Protocol = ""
      DefaultIcon
         (Default) = "alert.exe,1"
      shell
         open
            command
               (Default) = "C:\Program Files\Alert\alert.exe" "%1"

**Comment by marindobyns85-sudo** [comment]
> My backup process:
> 
> 1. Take screenshot of Google Authenticator export QR code.
> 2. Use cable to move it to PC.
> 3. Need to load it into program ...

This person is trying to hack my crypto
